src/code/z_sram_NES.c

```diff
@@ -1146,80 +1229,95 @@ void func_80145698(SramContext* sramCtx) {
 
             phi_s2 = false;
             sp6E = 0;
-            if (func_80185968(sramCtx->saveBuf, D_801C67C8[sp64], D_801C67F0[sp64])) {
+            // read main save from flash
+            if (func_80185968(sramCtx->saveBuf, gFlashSaveStartPages[sp64], gFlashSaveNumPages[sp64]) != 0) {
+                // main save didn't work
                 sp6E = 1;
-                if (func_80185968(sramCtx->saveBuf, D_801C67C8[sp64 + 1], D_801C67F0[sp64 + 1])) {
+                // read backup save from flash
+                if (func_80185968(sramCtx->saveBuf, gFlashSaveStartPages[sp64 + 1], gFlashSaveNumPages[sp64 + 1]) !=
+                    0) {
+                    // backup save didn't work
                     phi_s2 = true;
                 }
             }
 
             if (sp76 < 2) {
-                fileSelect->unk_24468[sp76] = 0;
+                // Non-owl save
+                // sp76 = 0: main save
+                // sp76 = 1: backup save
+
+                fileSelect->isOwlSave[sp76] = 0;
                 if (phi_s2) {
+                    // both main save and backup save failed
                     bzero(sramCtx->saveBuf, SAVE_BUFFER_SIZE);
-                    Lib_MemCpy(&gSaveContext, sramCtx->saveBuf, D_801C6870[sp64]);
+                    Lib_MemCpy(&gSaveContext, sramCtx->saveBuf, gFlashSaveSizes[sp64]);
                 } else {
-                    // Lib_MemCpy(&gSaveContext, sramCtx->saveBuf, D_801C6870[sp64]);
-                    Lib_MemCpy(&gSaveContext, sramCtx->saveBuf, D_801C6870[sp64]);
-                    temp_s2 = gSaveContext.save.checksum;
+                    Lib_MemCpy(&gSaveContext, sramCtx->saveBuf, gFlashSaveSizes[sp64]);
+
+                    // test checksum of main save
+                    oldCheckSum = gSaveContext.save.checksum;
                     gSaveContext.save.checksum = 0;
-                    temp_v0_2 = Sram_CalcChecksum(&gSaveContext, D_801C6870[sp64]);
-                    gSaveContext.save.checksum = temp_s2;
+                    newCheckSum = Sram_CalcChecksum(&gSaveContext, gFlashSaveSizes[sp64]);
+                    gSaveContext.save.checksum = oldCheckSum;
 
-                    if (CHECK_NEWF(gSaveContext.save.playerData.newf) || (temp_s2 != temp_v0_2)) {
+                    if (CHECK_NEWF(gSaveContext.save.playerData.newf) || (oldCheckSum != newCheckSum)) {
+                        // checksum didnt match, try backup save
                         sp6E = 1;
-                        if (CHECK_NEWF2(gSaveContext.save.playerData.newf)) {}
 
-                        phi_s2 = false;
-                        if (func_80185968(sramCtx->saveBuf, D_801C67C8[sp64 + 1], D_801C67F0[sp64 + 1])) {
+                        if ((gSaveContext.save.playerData.newf[0] == 'Z') &&
+                            (gSaveContext.save.playerData.newf[1] == 'E')) {
+                            phi_s2 = false;
+                        }
+
+                        // phi_s2 = false;
```

Comment:
Because this is still a non-equivalent function, I'm actually not sure where this is suppose to go. The original mips2c had it below, but the regalloc looks much better with it above